subsys/settings/src/settings_shell.c

```diff
@@ -106,19 +106,55 @@ static int cmd_read(const struct shell *shell_ptr, size_t argc, char *argv[])
 	err = settings_load_subtree_direct(name, settings_read_callback, &params);
 
 	if (err) {
-		shell_error(shell_ptr, "Failed to load settings: %d", err);
+		shell_error(shell_ptr, "Failed to load setting: %d", err);
 	} else if (!params.value_found) {
 		shell_error(shell_ptr, "Setting not found");
 	}
 
 	return 0;
 }
 
+static int cmd_write(const struct shell *shell_ptr, size_t argc, char *argv[])
+{
+	int err;
+	uint8_t buffer[CONFIG_SHELL_CMD_BUFF_SIZE / 2];
+	size_t buffer_len;
+
+	buffer_len = hex2bin(argv[2], strlen(argv[2]), buffer, sizeof(buffer));
+
+	if (buffer_len == 0) {
+		shell_error(shell_ptr, "Hex value has odd length");
```

Comment:
```suggestion
		shell_error(shell_ptr, "Failed to parse hex value");
```
This is a bit strange error message. 0 is returned in case of error, e.g. when string is not hex or something.

Reply:
@de-nordic Do you agree on that?

Reply:
Yeah. I do. Btw return the -EINVAL here, as you have previously did.